### ISSUE TITLE
Example SAML2 backend SP file configures a prohibited binding

### ISSUE THREAD

**Issue by alexstuart** [IdentityPython/SATOSA#224]
## Code Version
version 3.4.8

## Expected Behavior

When an Identity Provider issues a `<Response>` to the Service Provider during Web SSO, section 4.1.2 of [SAMLCore](http://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf) states that:

> Either the HTTP POST, or HTTP Artifact binding can be used to transfer the message to the service provider through the user agent. [...] The HTTP Redirect binding MUST NOT be used, as the response will typically exceed the URL length permitted by most user agents.

This implies that the configuration of the SP and the metadata describing its capabilities must not include an `AssertionConsumerService` endpoint with binding `urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect`.

## Current Behavior

The [example saml2_backend.yaml](https://github.com/IdentityPython/SATOSA/blob/04907b44dbe93eea2f46026e7b6923435accba5a/example/plugins/backends/saml2_backend.yaml.example#L48) file configures the incorrect binding:

```
          assertion_consumer_service:
          - [<base_url>/<name>/acs/post, 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST']
          - [<base_url>/<name>/acs/redirect, 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect']
```
and the generated backend metadata then includes an incorrect `AssertionConsumerService` endpoint. 

When you attempt to register the SP metadata with a federation that includes a comprehensive metadata checking framework, the SP metadata will be rejected. See, for example, [the UK federation's checking framework](https://github.com/ukf/ukf-meta/blob/master/mdx/_rules/check_bindings.xsl#L36) which does not allow the `HTTP-Redirect` binding.

## Possible Solution
The example file should not show how to configure the incorrect binding.

## Steps to Reproduce

1. Use `plugins/backends/saml2_backend.yaml` as a BACKEND_MODULES
1. Configure the endpoint as in the example file
1. Generate metadata using `satosa-saml-metadata`
1. Notice the incorrect `AssertionConsumerService` endpoint

**Comment by c00kiemon5ter** [comment]
I don't this quote in the linked document, neither in the bindings document. Other than that, I think what is said is reasonable. I will look into this in order to merge #225 and maybe also add a warning when that configuration is requested.

**Comment by alexstuart** [comment]
Sorry ... not sure why I linked the wrong doc!

**Comment by c00kiemon5ter** [comment]
fixed by #255 - commit 123620d4056032ff7f811ad1fb3078741df4dd33
